airbyte-integrations/connectors/source-zendesk-chat/build_customization.py

```diff
@@ -0,0 +1,21 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from dagger import Container
+
+MAIN_FILE_NAME = "main_dev.py"
+
+
+async def pre_connector_install(base_image_container: Container) -> Container:
+    """This function will run before the connector installation.
+    We set these environment variable to match what was originally in the Dockerfile.
+    Disclaimer: I have no idea if these env vars are actually needed.
+    """
+    return base_image_container.with_env_variable("AIRBYTE_IMPL_MODULE", "source_zendesk_chat").with_env_variable(
```

Comment:
I **think** I understand **why** this would fail if we had an await here.

As I understanding calling `await` on a `Container` evaluates the container.

_analogy: turns a docker file into an image_

so if we awaited here we would transform the return into something we could not keep adding container commands too like `with_label` `with_secret` etc...

Now I could not figure out why it would only error when publishing a amd64 docker image and not during build.....

That is the big mystery.

I think there's enough here to approve to unblock, but if you want to keep investigating on your side I wouldnt object!

Reply:
You stated it better than I did!
I originally called await to early evaluate the hooks instruction so that we can fail fast during the build if a hook is improperly set. 
`await` on a container evaluates its but also returns it so subsequent mutation on an awaited container work as long as you reassign it to a new variable.
I believe the problem lies in the fact that early evaluation messes something up with architecture variant... something like this container, built for a different platform in the first iteration of the build loop, is not recreated for the second platform on the second loop iteration...